duckduckgo/macos-browser
# Prevents VPNSettings from reporting fake changes

Task/Issue URL: https://app.asana.com/0/0/1206212872939261/f

iOS PR: https://github.com/duckduckgo/iOS/pull/2288
BSK PR: https://github.com/duckduckgo/BrowserServicesKit/pull/614

## Description

Makes some changes to VPNSettings so that it won't publish fake settings updates.

Also changes main App's startup behaviour slightly so that the VPN menu app will relaunch every time the main app launches (for debug builds only).  This is to make testing easier.

## Testing

1. Run `launchctl remove com.duckduckgo.macos.vpn.debug` from your terminal to ensure the VPN menu app isn't running.
2. In Xcode, select the DuckDuckGoVPN target, and run it.
3. The first time it launches, start the VPN
4. In Xcode, re-run the same target so it's launched again.
5. Ensure the VPN doesn't  disconnect when the app is launched again.

---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)
